include/git2/remote.h

```diff
@@ -562,6 +576,12 @@ struct git_remote_callbacks {
 	 * as the last parameter.
 	 */
 	void *payload;
+
+	/**
+	 * Resolve URL before connecting to remote.
+	 * The returned URL will be used to connect to the remote instead.
+	 */
+	git_resolve_url_cb resolve_url;
```

Comment:
I'd like this to be `git_url_resolve_cb`, since that sounds more like something that acts on (resolves) an object (url) instead of something that urls a resolve.

I'm otherwise :+1: on this, so I'm happy to make this change and merge it though to get this unblocked if you're busy.

Reply:
I think it sounds weird in any order, so I renamed. A few weeks later, I have the feeling it could be provided as some kind of "proxying" internal transport, but maybe that's too much (that would make this use case go through the already existing custom transport callback just above this one).